Source/ARTRealtime.m

```diff
@@ -776,20 +796,17 @@ - (void)onConnected:(ARTProtocolMessage *)message {
     
     // Resuming
     if (_resuming) {
-        if (![message.connectionId isEqualToString:self.connection.id_nosync]) { // RTN15c3
-            [self.logger warn:@"RT:%p connection \"%@\" has reconnected, but resume failed. Reattaching any attached channels", self, message.connectionId];
-            // Reattach all channels
-            for (ARTRealtimeChannelInternal *channel in self.channels.nosyncIterable) {
-                [channel reattachWithReason:message.error callback:nil];
-            }
-            _resuming = false;
-        }
-        else if (message.error) {
-            [self.logger warn:@"RT:%p connection \"%@\" has resumed with non-fatal error \"%@\"", self, message.connectionId, message.error.message];
-            // The error will be emitted on `transition`
-        }
-        else {
+        if ([message.connectionId isEqualToString:self.connection.id_nosync] && !message.error) {  // RTN15c6
+            [self reattachChannelsOnResumeResult:message];
             [self.logger debug:@"RT:%p connection \"%@\" has reconnected and resumed successfully", self, message.connectionId];
+            _resuming = false;
+        }else if(![message.connectionId isEqualToString:self.connection.id_nosync]){ //RTN15c7
+            if (message.error) {
+                [self.logger warn:@"RT:%p connection \"%@\" has resumed with non-fatal error \"%@\"", self, message.connectionId, message.error.message];
+            }
+            self.msgSerial = 0;
+            self.connection.latestMessageSerial = 0;
```

Comment:
This all seems like a near-duplicate of the `if (connIdChanged || recoverFailure) {` condition a few lines further down? Do we need both? (admittedly this block is only run if the `_resuming` bool is true, which I don't really understand)

Reply:
Andy also raised a similar comment [here](https://github.com/ably/ably-cocoa/pull/1496/files#r1000727312) .  TLDR - That `_resuming` flag is used locally to (presumably) differ between initial connections and subsequent ones. Block below doesn't take this into account and previously 'resume' rules were defined in this block - so I simply replaced it.  Also the condition here is slightly different than the one below see `recoverFailure`

```
 BOOL connIdChanged = prevConnId && ![message.connectionId isEqualToString:prevConnId];
            BOOL recoverFailure = !prevConnId && message.error;
            if (connIdChanged || recoverFailure) {
                [self.logger debug:@"RT:%p msgSerial of connection \"%@\" has been reset", self, self.connection.id_nosync];
                self.msgSerial = 0;
                self.connection.latestMessageSerial = 0;
                self.pendingMessageStartSerial = 0;
            }
```